core/finality-grandpa/src/communication/mod.rs

```diff
@@ -159,7 +163,11 @@ impl<B, S, H> Network<B> for Arc<NetworkService<B, S, H>> where
 		// waiting for the oneshot to resolve and from there on acting like a normal message channel.
 		let (tx, rx) = oneshot::channel();
 		self.with_gossip(move |gossip, _| {
-			let inner_rx = gossip.messages_for(GRANDPA_ENGINE_ID, topic);
+			let inner_rx = gossip
+				.messages_for(GRANDPA_ENGINE_ID, topic)
+				.map(|x| Ok(x))
+				.compat()
```

Comment:
If I have understood correctly, at some point when the call site of `gossip.message_for()` also returns a compatible future, this can+should be removed. If this is the case, you can mark `#[allow(deprecated)]` as `TODO: #3099 ...` or something similar.